base/threadingconstructs.jl

```diff
@@ -46,7 +56,12 @@ function _threadsfor(iter,lbody)
         ccall(:jl_threading_run, Void, (Any,), Core.svec($fun))
     end
 end
+"""
+`@threads`
+
+A macro to parallelize a for-loop to run with multiple threads. This spawns `nthreads()` number of threads, splits the iteration space amongst them, and iterates in parallel. A barrier is placed at the end of the loop which waits for all the threads to finish execution, and the loop returns.
 
+"""
```

Comment:
@vtjnash I didn't understand your question. Are you suggesting "... the threads to finish execution, and the loop returns." to be modified to "... the threads to finish execution before the loop returns." ?